Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs

```diff
@@ -91,17 +103,53 @@ private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
 
         _aliveNukeops.Clear();
 
-        // Between 1 and <max op count>: needs at least n players per op.
-        var numOps = Math.Max(1,
-            (int)Math.Min(
-                Math.Floor((double)ev.PlayerPool.Count / _cfg.GetCVar(CCVars.NukeopsPlayersPerOp)), _cfg.GetCVar(CCVars.NukeopsMaxOps)));
-        var ops = new IPlayerSession[numOps];
-        for (var i = 0; i < numOps; i++)
+        // Basically copied verbatim from traitor code
+        var playersPerOperative = _cfg.GetCVar(CCVars.NukeopsPlayersPerOp);
+        var maxOperatives = _cfg.GetCVar(CCVars.NukeopsMaxOps);
+
+        var list = new List<IPlayerSession>(ev.PlayerPool).Where(x =>
+            x.Data.ContentData()?.Mind?.AllRoles.All(role => role is not Job {CanBeAntag: false}) ?? false
+        ).ToList();
+        var prefList = new List<IPlayerSession>();
+        var operatives = new List<IPlayerSession>();
+
+        foreach (var player in list)
         {
-            ops[i] = _random.PickAndTake(ev.PlayerPool);
+            if (!ev.Profiles.ContainsKey(player.UserId))
+            {
+                continue;
+            }
+            var profile = ev.Profiles[player.UserId];
+            if (profile.AntagPreferences.Contains(NukeopsPrototypeId))
+            {
+                prefList.Add(player);
+            }
         }
 
-        var map = "/Maps/infiltrator.yml";
+        var numNukies = MathHelper.Clamp(ev.PlayerPool.Count / playersPerOperative, 1, maxOperatives);
+
+        for (var i = 0; i < numNukies; i++)
+        {
+            IPlayerSession nukeOp;
+            if (prefList.Count == 0)
+            {
+                if (list.Count == 0)
+                {
+                    Logger.InfoS("preset", "Insufficient ready players to fill up with nukeops, stopping the selection");
+                }
+                nukeOp = _random.PickAndTake(list);
+                Logger.InfoS("preset", "Insufficient preferred nukeops, picking at random.");
+            }
+            else
+            {
+                nukeOp = _random.PickAndTake(prefList);
+                list.Remove(nukeOp);
+                Logger.InfoS("preset", "Selected a preferred nukeop.");
+            }
+            operatives.Add(nukeOp);
+        }
+
+        var map = _config.Shuttles != null ? _random.Pick(_config.Shuttles) : "Maps/infiltrator.yml";
```

Comment:
just give the datafield a default value with this map instead of doing that logic here